### PR TITLE
Interface triggers 83

### PR DESCRIPTION
**Pull request by russelldutton**
## TL;DR
The current state of my work on porting the team desires customization, which was previously sliders, to dynamic components with drop-downs for more control and specialisation in its own page.

## Details & Current Functionality
I created a new Component: TeamDesiresComponent, to contain the temporary code for porting the desires' configuration. I also created a service responsible for the creation and retrieval of the ConfigurationFormat objects.
Functionality:
- Can dynamically add/remove conditions, and per lane where applicable.
- Service currently only gets an empty ConfigurationFormat object as I have yet to mock the retrieval in the backend (See TODO), but the object is retrieved from a service.
- Can dynamically add groups of conditions, which correspond to CompoundConditions.
- Grouped viewable options to customize by function (push, defend, farm, roshan, roam) i.e. you only view Roshan Customisation options if it's selected, lowering information overload

## TODO
- Dynamically Remove ConditionGroups (CompoundConditions)
- Mock retrieval of config format object from backend

### CODE REVIEW

**Comment by SpellBreakerZA** [comment]
Hey, I was just going over the component for interest sake. 
I know that this is not the final design but I have a few suggestions.

- Instead of having a mid, bot and top tab, could you not rather have an image of the map and then select the lane and have it be highlighted (like the map we had on the old config page)?
It will be easier to understand and look fancier too :smile: 

- Also regarding the operator maybe we should have the mathematical symbol and then in brackets have the meaning. I find that the maths is easier to read at a glance:
"≤ (less than or equal to)"

- Side note: the inputs are just type text atm, I can enter ≤ "abcc". It looks like they'll mainly be numeric so that will be a quick change
